### PR TITLE
feat: guest cart loading improvement

### PR DESCRIPTION
**Pull request by dunqan**
- drops AnonAuthTokenService and AnonTokenInterceptors
- handles guest user in IdentityService with optional requireGuest flag
- add GuestIdentityInterceptor
- guest user Id is not created and guest cart is not fetched before first add to cart

Closes [HRZ-90591](https://spryker.atlassian.net/browse/HRZ-90591)

[HRZ-90591]: https://spryker.atlassian.net/browse/HRZ-90591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/auth/src/sapi/sapi-identity.service.ts [hunk 1]
```diff
@@ -2,31 +2,81 @@ import {
...
 import { parseToken } from './utils';
 
 export class SapiIdentityService implements IdentityService {
+  protected ANONYMOUS_USER_IDENTIFIER = 'oryx.anonymous-user';
+
+  protected generateGuest$ = new BehaviorSubject<boolean>(false);
+
   protected identity$ = this.authService.getToken().pipe(
-    map((token) => this.getFromToken(token)),
+    featureVersion >= '1.3'
```
Comment (tolerants) [inline libs/platform/auth/src/sapi/sapi-identity.service.ts]:
```suggestion
    featureVersion >= '1.4'
```

File: libs/platform/auth/src/services/anon-auth-token.service.ts [hunk 2]
```diff
@@ -41,7 +43,11 @@ export class AnonAuthTokenService implements AuthTokenService {
       .get<string>(this.ANONYMOUS_USER_IDENTIFIER, StorageType.Session)
       .pipe(
         switchMap((userId) => {
-          return userId ? of(userId) : this.createAnonymousId();
+          return featureVersion >= '1.3'
```
Comment (tolerants) [inline libs/platform/auth/src/services/anon-auth-token.service.ts]:
```suggestion
          return featureVersion >= '1.4'
```

**Review by tolerants — CHANGES_REQUESTED** [review]

File: libs/platform/auth/src/services/anon-auth-token.service.ts [hunk 3]
```diff
@@ -41,7 +43,11 @@ export class AnonAuthTokenService implements AuthTokenService {
       .get<string>(this.ANONYMOUS_USER_IDENTIFIER, StorageType.Session)
       .pipe(
         switchMap((userId) => {
-          return userId ? of(userId) : this.createAnonymousId();
+          return featureVersion >= '1.4'
+            ? of(userId)
+            : userId
+            ? of(userId)
+            : this.createAnonymousId();
```
Comment (tobi-or-not-tobi) [inline libs/platform/auth/src/services/anon-auth-token.service.ts]:
```suggestion
          return featureVersion >= '1.4' || userId
            ? of(userId)
            : this.createAnonymousId();
```

Comment (tobi-or-not-tobi) [reply libs/platform/auth/src/services/anon-auth-token.service.ts]: but seems that we do not need this, as we're no longer using this service in >= 1.4

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/platform/auth/src/services/anon-auth-token.service.ts (hunk 3 above)
Comment (dunqan) [reply libs/platform/auth/src/services/anon-auth-token.service.ts]: 👍

**Review by tobi-or-not-tobi — DISMISSED** [review]

File: libs/platform/auth/src/sapi/feature.ts [hunk 4]
```diff
@@ -92,7 +109,9 @@ export class SapiAuthFeature extends OauthFeature implements AppFeature {
 
 export interface SapiAuthFeatureConfig extends OauthFeatureConfig {
   skipRoutes?: boolean;
+  /** deprecated since 1.4 */
```
Comment (tolerants) [inline libs/platform/auth/src/sapi/feature.ts]: Maybe let's add a note that `guestIdentityInterceptor` should be used instead, so it's clear without taking a look into this interface.

File: libs/platform/auth/src/services/anon-auth-token.service.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ import { AuthTokenData, AuthTokenService } from './auth-token.service';
 import { AuthService } from './auth.service';
 import { generateID } from './utils';
 
+/** deprecated since 1.4 */
```
Comment (tolerants) [inline libs/platform/auth/src/services/anon-auth-token.service.ts]: Suggest to add a replacement.

**Review by tolerants — APPROVED** [review]

**Pull request merged**